File: Nafza-Faroidh-CS/PrintJob.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Drawing.Printing;
using System.Drawing;

public partial class Form1
{


	int Queried;
	int QueriedPage;
	float PanjangLaman;
	Point PosisiLaman;
	Font f;

	const int  MaxListPerPage = 25;
	private void PrintDocument1_BeginPrint(object sender, System.Drawing.Printing.PrintEventArgs e)
	{
		Queried = 0;
		QueriedPage = 0;
		f = new Font("Times New Roman", 12);
	}

	private void PrintDocument1_PrintPage(System.Object sender, PrintPageEventArgs e)
	{
		e.PageSettings.Margins = new Margins(50, 50, 100, 100);
		var m = e.PageSettings.Margins;
		PosisiLaman = new Point(m.Left, m.Top);
		PanjangLaman = e.PageSettings.PrintableArea.Width - m.Left - m.Right;

        QueriedPage = 0;
		DrawHeader(e.Graphics);
		for (int index = 0; index <= MaxListPerPage; index++) {
			if (Helper.anggotas.Count >= Queried + QueriedPage * MaxListPerPage)
				break; 
			DrawPerAnggota(e.Graphics);
			Queried += 1;
		}
		e.HasMorePages = Helper.anggotas.Count > (QueriedPage + 1) * MaxListPerPage;
	}

	public void DrawPerAnggota(Graphics g)
	{
		dynamic idx = Queried + QueriedPage * MaxListPerPage;
		dynamic a = Helper.anggotas[idx];
		dynamic b = new SolidBrush(Color.Black);
		int r = 0;
		Math.DivRem(idx, 2, out r);
		if (r == 0) {
			g.FillRectangle(new SolidBrush(Color.WhiteSmoke), new Rectangle(LokasiAngka(), new Size((int)PanjangLaman, 20)));
		}
		g.DrawString((idx + 1).ToString(), f, b, LokasiAngka());
		g.DrawString(a.nama, f, b, LokasiNama());
		g.DrawString(Helper.Anggota2String(a.tipe), f, b, LokasiTipe());
		g.DrawString(Helper.Jatah2String(a.hasil_jatah), f, b, LokasiJatah());
		g.DrawString("Rp. " + a.hasil_waris.ToString("##,#0"), f, b, LokasiWaris());

	}

	public void DrawHeader(Graphics g)
	{
		Queried = -2;
		dynamic b = new SolidBrush(Color.Black);
		g.DrawImage(Faroidh.Properties.Resources.HeaderImg, PosisiLaman.X, Convert.ToInt32(PosisiLaman.Y - PanjangLaman * 0.121 + 50), Convert.ToInt32(PanjangLaman), Convert.ToInt32(PanjangLaman * 0.121));
		g.FillRectangle(new SolidBrush(Color.DarkGray), new Rectangle(PosisiLaman.X, PosisiLaman.Y + 50, (int)PanjangLaman, 40));
		g.DrawString("NO", f, b, LokasiAngka());
		g.DrawString("NAMA", f, b, LokasiNama());
		g.DrawString("STATUS", f, b, LokasiTipe());
		g.DrawString("BAGIAN", f, b, LokasiJatah());
		g.DrawString("JATAH", f, b, LokasiWaris());
		g.DrawLine(new Pen(b), PosisiLaman.X, PosisiLaman.Y + 90, PosisiLaman.X + PanjangLaman, PosisiLaman.Y + 90);
		g.DrawLine(new Pen(b), PosisiLaman.X, PosisiLaman.Y + 50, PosisiLaman.X + PanjangLaman, PosisiLaman.Y + 50);
		Queried = 0;
	}

	public Point LokasiAngka()
	{
		return new Point((int)(PosisiLaman.X + PanjangLaman * 0.00), PosisiLaman.Y + Queried * 20 + 100);
	}

	public Point LokasiNama()
	{
		return new Point((int)(PosisiLaman.X + PanjangLaman * 0.05), PosisiLaman.Y + Queried * 20 + 100);
	}

	public Point LokasiTipe()
	{
		return new Point((int)(PosisiLaman.X + PanjangLaman * 0.25), PosisiLaman.Y + Queried * 20 + 100);
	}

	public Point LokasiJatah()
	{
		return new Point((int)(PosisiLaman.X + PanjangLaman * 0.6), PosisiLaman.Y + Queried * 20 + 100);
	}

	public Point LokasiWaris()
	{
		return new Point((int)(PosisiLaman.X + PanjangLaman * 0.8), PosisiLaman.Y + Queried * 20 + 100);
	}


}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================

File: Nafza-Faroidh-VB/PrintJob.vb
Imports System.Drawing.Printing

Partial Public Class Form1

    Dim Queried As Integer
    Dim QueriedPage As Integer
    Dim PanjangLaman As Decimal
    Dim PosisiLaman As Point
    Dim f As Font
    Const MaxListPerPage = 25

    Private Sub PrintDocument1_BeginPrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles PrintDoc.BeginPrint
        Queried = 0
        QueriedPage = 0
        f = New Font("Times New Roman", 12)
    End Sub

    Private Sub PrintDocument1_PrintPage(ByVal sender As System.Object, ByVal e As PrintPageEventArgs) Handles PrintDoc.PrintPage
        e.PageSettings.Margins = New Margins(50, 50, 100, 100)
        Dim m = e.PageSettings.Margins
        PosisiLaman = New Point(m.Left, m.Top)
        PanjangLaman = e.PageSettings.PrintableArea.Width - m.Left - m.Right
        'MyBase.OnPrintPage(e)
        'QueriedPage += 1
        QueriedPage = 0
        DrawHeader(e.Graphics)
        For index = 0 To MaxListPerPage
            If Helper.anggotas.Count >= Queried + QueriedPage * MaxListPerPage Then Exit For
            DrawPerAnggota(e.Graphics)
            Queried += 1
        Next
        e.HasMorePages = Helper.anggotas.Count > (QueriedPage + 1) * MaxListPerPage
    End Sub

    Sub DrawPerAnggota(g As Graphics)
        Dim idx = Queried + QueriedPage * MaxListPerPage
        Dim a = Helper.anggotas(idx)
        Dim b = New SolidBrush(Color.Black)
        Dim r As Integer
        Math.DivRem(idx, 2, r)
        If r = 0 Then
            g.FillRectangle(New SolidBrush(Color.WhiteSmoke), New Rectangle(LokasiAngka(), New Size(PanjangLaman, 20)))
        End If
        g.DrawString(idx + 1, f, b, LokasiAngka())
        g.DrawString(a.nama, f, b, LokasiNama())
        g.DrawString(Helper.Anggota2String(a.tipe), f, b, LokasiTipe())
        g.DrawString(Helper.Jatah2String(a.hasil_jatah), f, b, LokasiJatah())
        g.DrawString("Rp. " + a.hasil_waris.ToString("##,#0"), f, b, LokasiWaris())

    End Sub

    Sub DrawHeader(g As Graphics)
        Queried = -2
        Dim b = New SolidBrush(Color.Black)
        g.DrawImage(My.Resources.HeaderImg, PosisiLaman.X, CInt(PosisiLaman.Y - PanjangLaman * 0.121 + 50), CInt(PanjangLaman), CInt(PanjangLaman * 0.121))
        g.FillRectangle(New SolidBrush(Color.DarkGray), New Rectangle(PosisiLaman.X, PosisiLaman.Y + 50, PanjangLaman, 40))
        g.DrawString("NO", f, b, LokasiAngka())
        g.DrawString("NAMA", f, b, LokasiNama())
        g.DrawString("STATUS", f, b, LokasiTipe())
        g.DrawString("BAGIAN", f, b, LokasiJatah())
        g.DrawString("JATAH", f, b, LokasiWaris())
        g.DrawLine(New Pen(b), PosisiLaman.X, PosisiLaman.Y + 90, PosisiLaman.X + PanjangLaman, PosisiLaman.Y + 90)
        g.DrawLine(New Pen(b), PosisiLaman.X, PosisiLaman.Y + 50, PosisiLaman.X + PanjangLaman, PosisiLaman.Y + 50)
        Queried = 0
    End Sub

    Function LokasiAngka() As Point
        Return New Point(PosisiLaman.X + PanjangLaman * 0.0, PosisiLaman.Y + Queried * 20 + 100)
    End Function

    Function LokasiNama() As Point
        Return New Point(PosisiLaman.X + PanjangLaman * 0.05, PosisiLaman.Y + Queried * 20 + 100)
    End Function

    Function LokasiTipe() As Point
        Return New Point(PosisiLaman.X + PanjangLaman * 0.25, PosisiLaman.Y + Queried * 20 + 100)
    End Function

    Function LokasiJatah() As Point
        Return New Point(PosisiLaman.X + PanjangLaman * 0.6, PosisiLaman.Y + Queried * 20 + 100)
    End Function

    Function LokasiWaris() As Point
        Return New Point(PosisiLaman.X + PanjangLaman * 0.8, PosisiLaman.Y + Queried * 20 + 100)
    End Function


End Class
